File: Lib/Telerik/Examples/QuickStart/TreeView/Globalization/Localization/GermanTreeViewLocalizationProvider.vb
Imports System.Text
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.TreeView.Globalization.Localization
	Public Class GermanTreeViewLocalizationProvider
		Inherits TreeViewLocalizationProvider
		Public Overrides Function GetLocalizedString(ByVal id As String) As String
            Select Case id
                Case TreeViewStringId.ContextMenuCollapse
                    Return "Zusammenklappen"
                Case TreeViewStringId.ContextMenuDelete
                    Return "Löschen"
                Case TreeViewStringId.ContextMenuEdit
                    Return "Edieren"
                Case TreeViewStringId.ContextMenuExpand
                    Return "Expandieren"
                Case TreeViewStringId.ContextMenuNew
                    Return "Neu"
                Case TreeViewStringId.ContextMenuCopy
                    Return "Kopie"
                Case TreeViewStringId.ContextMenuPaste
                    Return "Shpacieren"
                Case TreeViewStringId.ContextMenuCut
                    Return "Schneiden"
            End Select

			Return ""
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/TreeView/Globalization/Localization/GermanTreeViewLocalizationProvider.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.TreeView.Globalization.Localization
{
    public class GermanTreeViewLocalizationProvider: TreeViewLocalizationProvider
    {
        public override string GetLocalizedString(string id)
        {
            switch (id)
            {
                case TreeViewStringId.ContextMenuCollapse:
                    return "Zusammenklappen";
                case TreeViewStringId.ContextMenuDelete:
                    return "Löschen";
                case TreeViewStringId.ContextMenuEdit:
                    return "Edieren";
                case TreeViewStringId.ContextMenuExpand:
                    return "Expandieren";
                case TreeViewStringId.ContextMenuNew:
                    return "Neu";
                case TreeViewStringId.ContextMenuCopy:
                    return "Kopie";
                case TreeViewStringId.ContextMenuPaste:
                    return "Shpacieren";
                case TreeViewStringId.ContextMenuCut:
                    return "Schneiden";
            }

            return "";
        }
    }
}
